### ISSUE TITLE
Load testing MyLA 

### ISSUE THREAD

**Issue by pushyamig** [tl-its-umich-edu/my-learning-analytics#297]
I think for next phase if we are opening to far more courses this should be on our Todo list.

**Comment by jonespm** [comment]
I'd also think we should look into implementing caching. Though if you having a "baseline" load test, then implement caching then another load test seems fine too.

**Comment by pushyamig** [comment]
Good Idea, What kind of caching you are thinking of DB caching, Memcaching etc

**Comment by jonespm** [comment]
Yeah probably memcached like this
https://docs.djangoproject.com/en/1.11/topics/cache/

And also if we ever have server side state we can store it there too so we can scale out. 

I think all we'd have to do is start up a new pod with memcached on it. Can probably just use the docker image! https://hub.docker.com/_/memcached/

**Comment by jennlove-um** [comment]
This should include application and database load testing.

**Comment by jonespm** [comment]
I have ran preliminary load tests on MyLA test and checked the code into Github at https://github.com/tl-its-umich-edu/tl-load-tests

Summary: It looks like we were able to handle about 250 users per 2 core server and 500 users max on 2 servers. I slowly ramped up the users (1 user per second up to 500) with response times constant until 500. 

At 500 users I added 2 more pods to the autoscaler bringing up the cluster capacity to 4 pods However performance was still unstable over 500 users

I'm not sure if the database was a bottleneck or something else, going to wait on response from database

The calls to the assignment and resources_api took about the same amount of time and 3 times slower than any other call. 

To go over 500 we’d probably need to

1. Look to see if debug the logging is having an adverse effect (it’s logging a ton of stuff)
2. See if the database is hitting capacity and if we can increase that. I can put a ticket in to investigate
3. Add in caching and/or performance improvements over the next few weeks/months
